demo/midgard-sdk/src/tx-builder/user-events/tx-order.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  CML,
+  Data,
+  LucidEvolution,
+  Script,
+  toUnit,
+  TxBuilder,
+  UTxO,
+} from "@lucid-evolution/lucid";
+import { POSIXTime, POSIXTimeSchema } from "@/tx-builder/common.js";
+import {
+  HashingError,
+  LucidError,
+  hashHexWithBlake2b256,
+} from "@/utils/common.js";
+import { TxOrderEventSchema } from "@/tx-builder/ledger-state.js";
+import { Effect } from "effect";
+
+export type TransactionOrderParams = {
+  txOrderAddress: string;
+  mintingPolicy: Script;
+  policyId: string;
+  refundAddress: string;
+  refundDatum: string;
+  inclusionTime: POSIXTime;
+  midgardTxBody: string;
+  midgardTxWits: string;
+};
```

Comment:
We should aim for having as much help as possible from the compiler. The types you've defined in `ledger-state` should be usable here.

Also hashing of the underlying values is something this function can handle, so it might be best if this type carries e.g. `MidgardTxBodyCompact` instead of its hash.